packages/moleculer-db-adapter-mongoose/src/index.js

```diff
@@ -311,13 +368,8 @@ class MongooseDbAdapter {
 	 * @memberof MongooseDbAdapter
 	 */
 	entityToObject(entity, ctx) {
-		const fieldsToPopulate = _.get(ctx, "params.populate", []);
-		const virtualFields = Object.values(_.get(this, "model.schema.virtuals", {}))
-			.reduce((acc, virtual) => _.get(virtual, "options.ref") ? [...acc, virtual.path] : acc, []);
-		const virtualsToPopulate = _.intersection(fieldsToPopulate, virtualFields);
-		const options = {skipInvalidIds: true, lean: true};
-		const transform = (doc) => doc._id;
-		const populate = virtualsToPopulate.map(path => ({path, select: "_id", options, transform}));
+		const useNativeMongooseVirtuals = _.get(ctx, "service.settings.useNativeMongooseVirtuals", false);
```

Comment:
There is a pointer to the service instace as `this.service`. Please use it instead of ctx.service...

In addition, this property is very constant, so better if you get it in `init` to a local class variable because lodash `get` is not too fast.

Reply:
I removed unnecessary use of `_.get`, it should be ok now.